okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java

```diff
@@ -731,12 +736,20 @@ private static String host(String input, int pos, int limit) {
       return idnDecoded;
     }
 
-    private static String decodeIpv6(String input, int pos, int limit) {
-      return input.substring(pos, limit); // TODO(jwilson) implement IPv6 decoding.
+    private static InetAddress decodeIpv6(String input, int pos, int limit) {
+      try {
+        return InetAddress.getByName(input.substring(pos, limit));
```

Comment:
Move substring to call site? Unless there's some other value I'm not seeing (on mobile).